tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/dav/DavClient.java

```diff
@@ -0,0 +1,307 @@
+            .request(HttpMethod.valueOf("REPORT"))
+            .uri(calendarURI.getPath())
+            .send(Mono.just(new GetCalendarByEventIdRequestBody(eventUid).asByteBuf()))
+            .responseSingle(
+                (response, responseContent) -> handleCalendarResponse(responseContent, response.status(), eventUid));
+    }
+
+    private Mono<DavCalendarObject> handleCalendarResponse(ByteBufMono responseContent, HttpResponseStatus responseStatus, String eventUid) {
+        if (responseStatus == HttpResponseStatus.MULTI_STATUS) {
+            return responseContent.asString(StandardCharsets.UTF_8)
+                .map(content -> XMLUtil.parse(content, DavMultistatus.class))
+                .map(this::extractCalendarObject)
+                .flatMap(Mono::justOrEmpty);
+        }
+        return Mono.error(new DavClientException(
+            String.format("Unexpected status code: %d when finding VCALENDAR object containing event: %s",
+                responseStatus.code(), eventUid)));
+    }
+
+    private Optional<DavCalendarObject> extractCalendarObject(DavMultistatus multistatus) {
+        return multistatus.getResponses()
+            .stream()
+            .findFirst()
+            .flatMap(DavCalendarObject::fromDavResponse);
+    }
+
+    public Flux<URI> findUserCalendars(String userId, String username) {
+        return client.headers(headers ->
+                headers.add(HttpHeaderNames.ACCEPT, ACCEPT_XML)
+                    .add(HttpHeaderNames.AUTHORIZATION,
+                        HttpUtils.createBasicAuthenticationToken(createDelegatedCredentials(username))))
+            .request(HttpMethod.valueOf("PROPFIND"))
+            .uri("/calendars/" + userId)
+            .responseSingle((response, byteBufMono) -> {
+                if (response.status() == HttpResponseStatus.MULTI_STATUS) {
+                    return byteBufMono.asString(StandardCharsets.UTF_8)
+                        .map(multiStatusResponse -> XMLUtil.parse(multiStatusResponse, DavMultistatus.class))
+                        .map(this::extractCalendarURIsFromResponse);
+                } else {
+                    return Mono.error(new DavClientException(
+                        String.format("Unexpected status code: %d when finding user calendars for user: %s",
+                            response.status().code(), userId)));
+                }
+            })
+            .flatMapMany(Flux::fromIterable);
+    }
+
+    private List<URI> extractCalendarURIsFromResponse(DavMultistatus multistatus) {
+        List<URI> hrefs = new ArrayList<>();
+
+        for (DavResponse response : multistatus.getResponses()) {
+            response.getHref().getValue().ifPresent(href -> {
+                if (CALENDAR_URI_PATTERN.matcher(href).matches() &&
+                    !(href.endsWith("inbox/") || href.endsWith("outbox/"))) {
+                    hrefs.add(URI.create(href));
```

Comment:
What happen if one URL out of 5 is invalid and throws upon parsing?

We aboort the whole sequence and crash?